src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8050,15 +8070,15 @@ bool tryLookupLengthOrCount(string propertyName, out PropertySymbol valid)
                     LookupOptions.Default,
                     originalBinder: this,
                     diagnose: false,
-                    useSiteInfo: ref discardedUseSiteInfo);
+                    useSiteInfo: ref useSiteInfo);
```

Comment:
I'm worried about the use-site info here. 
Consider if `LookupMembersInType` adds some use-site info, but then we reject that candidate in the `if` below. 
Then would the second call to `tryLookupLengthOrCount` (line 8060 above) clear that use-site info?

I suspect we may have to use a temporary `CompoundUseSiteInfo` instead, and merge  it once we decide we'll use the member we found.

Reply:
Sounds like a good idea. I'll be fine to do this for the new PerformPatternIndexerLookup but this is existing code that I don't want to change, unless it won't affect range indexer binding?

Reply:
Actually, it looks like we only call `TryLookupLengthOrCount` with `CompoundUseSiteInfo<AssemblySymbol>.Discarded`, so we're fine here. Thanks
nit: consider adding an assertion to that effect then.